### PR TITLE
add song feature dialog when clicking flower petals

### LINKED ISSUE #117: Song Features Timeline
Click on petal and modal shows up with more detail of the exact values of songs' features.

### PR DESCRIPTION
**Pull request by danielamormocea**
Fixes #117

https://github.com/user-attachments/assets/ecf59be4-6a57-4bcf-af64-25be93ca4683

### CODE REVIEW

File: server/app/controller.py
Comment (jemaie) [inline server/app/controller.py:1]: please remove print debug statement :)

File: server/app/service.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ async def get_song_from_database(self, song_id: str) -> Song:
 
     async def get_matching_songs_from_database(self, pattern: str, limit: int) -> SongList:
         result = await self.repo.get_songs_by_pattern(pattern, limit)
-        songs = [Song.model_validate_json(row) for row in result]
+        songs = [Song.parse_obj(dict(row)) for row in result]
```
Comment (maettuu) [inline server/app/service.py:237]: can you let me know why this change please? :) thx

**Comment by saeie07** [comment]
video looks great. I think what I would adapt is, right now all features have the same color. this should change, so it matches. Energy = Blue, Danceability = Green, Speechiness = Orange, etc.

File: server/app/service.py (hunk 1 above)
Comment (danielamormocea) [reply server/app/service.py:237]: Because the method does not work with the frontend otherwise. I receive an error as it's unable to parse the object.

Comment (jemaie) [reply server/app/service.py:237]: Could you shortly investigate if there isn't an issue with the data or something else? Is it a ValidationError?

"model_validate_json" returns a validated pydantic model. If we don't validate it, I fear issues down the path ..
Also, parse_obj is deprecated from Pydantic V1. V2 encourages model_validate() otherwise.
Therefore, maybe model_validate() could work instead of model_validate_json().

Comment (danielamormocea) [reply server/app/service.py:237]: Sure, I would suggest to do this in a follow-up, to not block this whole PR for now. I can add a TODO and I'll add a task for it. :)

**Review by jemaie — APPROVED** [review]
